proto/buf/alpha/registry/v1alpha1/labels.proto

```diff
@@ -0,0 +1,65 @@
+// Copyright 2020-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package buf.alpha.registry.v1alpha1;
+
+import "google/protobuf/timestamp.proto";
+
+enum LabelNamespace {
+  LABEL_NAMESPACE_UNSPECIFIED = 0;
+  LABEL_NAMESPACE_TAG = 1;
+  LABEL_NAMESPACE_BRANCH = 2;
+  LABEL_NAMESPACE_GITCOMMIT = 3;
+}
+
+message Label {
+  LabelNamespace namespace = 1;
+  string name = 2;
+  string commit_id = 3;
+}
+
+service LabelService {
+  rpc CreateLabel(CreateLabelRequest) returns (CreateLabelResponse);
+  rpc MoveLabel(MoveLabelRequest) returns (MoveLabelResponse);
+  rpc GetLabelByNameAndNamespace(GetLabelByNameAndNamespaceRequest) returns (GetLabelByNameAndNamespaceResponse);
+}
+
+message CreateLabelRequest {
+  Label label = 1;
+  string author = 2;
+  google.protobuf.Timestamp create_time = 3;
+}
+
+message CreateLabelResponse {
+  string commit_id = 1;
+}
+
+message MoveLabelRequest {
+  Label label = 1;
+  string from = 2;
+  string to = 3;
+}
+
+message MoveLabelResponse {}
+
+message GetLabelByNameAndNamespaceRequest {
+  string name = 1;
+  LabelNamespace namespace = 2;
```

Comment:
My original thought here was that if we set the "all" or "unspecified" namespace for querying, it would query across namespaces. But to Tristan's point above, if we're only querying by name, then maybe this is moot.